http/src/main/java/io/micronaut/http/bind/binders/QueryValueArgumentBinder.java

```diff
@@ -130,6 +146,51 @@ public BindingResult<T> bind(ArgumentConversionContext<T> context, HttpRequest<?
         return doBind(context, parameters, BindingResult.unsatisfied());
     }
 
+    private BindingResult<T> bindPojo(ArgumentConversionContext<T> context,
+                                      HttpRequest<?> source,
+                                      AnnotationMetadata annotationMetadata,
+                                      ConvertibleMultiValues<String> parameters,
+                                      Argument<T> argument) {
+        Optional<BeanIntrospection<T>> introspectionOpt = BeanIntrospector.SHARED.findIntrospection(argument.getType());
+        if (introspectionOpt.isEmpty()) {
+            return BindingResult.unsatisfied();
+        }
+
+        BeanIntrospection<T> introspection = introspectionOpt.get();
+        BeanIntrospection.Builder<T> introspectionBuilder = introspection.builder();
+
+        for (BeanProperty<T, Object> property : introspection.getBeanProperties()) {
```

Comment:
iterative here over the [getBuilderArguments](https://docs.micronaut.io/latest/api/io/micronaut/core/beans/BeanIntrospection.Builder.html#getBuilderArguments())() using array iteration to obtain the index